### DISCUSSION TITLE
Meaining of 'poetry show' colors

### DISCUSSION THREAD

**Discussion by yrro** [python-poetry/poetry#6330] [Q&A] [answered]
Is there any documentation for the meaning of the colours that dependencies are shown in?

![image](https://user-images.githubusercontent.com/309468/187923108-05bb939a-689e-4613-8dca-734c16e64071.png)

"blue package, red version" = semver-safe update available?
"red package, red version" = ?
"blue package, yellow version" = update possible but not semver-safe?

And/or a CLI flag to have this information output in an accessibility-friendly way?

**Comment by yrro** [comment, marked as answer]
Answering my own question after looking at the source

red package = package isn't currently installed
blue package = package is installed

red version = a semver-compatible update is available
yellow version = updates are available but are not semver-compatible

**Reply by deadnews** [reply]
Yes.

Code ref:

https://github.com/python-poetry/poetry/blob/81c7d00cb3985c6db81d8370ef400731b05bf047/src/poetry/console/commands/show.py#L308-L309

https://github.com/python-poetry/poetry/blob/81c7d00cb3985c6db81d8370ef400731b05bf047/src/poetry/console/commands/show.py#L337-L340
